lib/spack/spack/fetch_strategy.py

```diff
@@ -453,35 +461,38 @@ def fetch(self):
 
 
 class VCSFetchStrategy(FetchStrategy):
+    """Superclass for version control system fetch strategies.
+
+    Like all fetchers, VCS fetchers are identified by the attributes
+    passed to the ``version`` directive.  The optional_attrs for a VCS
+    fetch strategy represent types of revisions, e.g. tags, branches,
+    commits, etc.
+
+    The required attributes (git, svn, etc.) are used to specify the URL
+    and to distinguish a VCS fetch strategy from a URL fetch strategy.
+
+    """
 
-    def __init__(self, name, *rev_types, **kwargs):
+    def __init__(self, **kwargs):
```

Comment:
It would be great if all of these kwargs were documented. But this isn't true in most of Spack 😢

Reply:
for the fetch strategies they're basically whatever comes from `version()`.  They're written as `**kwargs` so that they're easy to delegate.  To some extent, they're more documented now -- just look at `optional_attrs`.